src/gnark_backend_wrapper/groth16/serialize.rs

```diff
@@ -1,73 +1,90 @@
-use super::acir_to_r1cs::{RawGate, RawR1CS};
-use acvm::acir::native_types::Witness;
+use super::acir_to_r1cs::{AddTerm, MulTerm};
+use crate::gnark_backend_wrapper as gnark_backend;
+use ark_serialize::CanonicalSerialize;
 use serde::{ser::SerializeStruct, Serialize};
 
-impl Serialize for RawR1CS {
-    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+impl Serialize for MulTerm {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: serde::Serializer,
     {
-        let mut s = serializer.serialize_struct("RawR1CS", 4)?;
+        let mut serialized_coefficient = Vec::new();
+        self.coefficient
+            .serialize_uncompressed(&mut serialized_coefficient)
+            .map_err(serde::ser::Error::custom)?;
+        // Turn little-endian to big-endian.
+        serialized_coefficient.reverse();
+        let encoded_coefficient = hex::encode(serialized_coefficient);
 
-        let mut serializable_values: Vec<String> = Vec::new();
-        for value in &self.values {
-            let mut serialized_value = Vec::new();
-            ark_serialize::CanonicalSerialize::serialize_uncompressed(value, &mut serialized_value)
-                .map_err(|e| serde::ser::Error::custom(e.to_string()))?;
-            serializable_values.push(hex::encode(serialized_value));
-        }
-
-        s.serialize_field("gates", &self.gates)?;
-        s.serialize_field("public_inputs", &self.public_inputs)?;
-        s.serialize_field("values", &serializable_values)?;
-        s.serialize_field("num_variables", &self.num_variables)?;
+        let mut s = serializer.serialize_struct("MulTerm", 3)?;
```

Comment:
What is the 3?

Reply:
`3` is the number of fields in the struct to serialize (`MulTerm` in this case). I'll encapsulate the number in a variable.